src/components/Table/Table.jsx

```diff
@@ -178,7 +180,7 @@ const propTypes = {
 };
 
 export const defaultProps = baseProps => ({
-  id: 'Table',
+  id: `Table_${tableIdNumber}`,
```

Comment:
This incrementing counter feels odd. I don't think it's necessary as consumers (and our stories) should configure a tableID, although it's not considered a required prop. 

Should we just use `Date.now()`?

Reply:
> I think something is up with the merge from resolving the conflicts - there are now lots of unrelated changes in here @davidicus @sstone2423
> 
> RowActionsCell, Table 'onColumnResize callback' story, ComposedModal, TableHead, columnWidthUtilityFunctions, etc

It looks like that charting changes are correct here. The diff is the same as master for the obvious ones that I know in BarChartCard. Could we close and reopen this, or reset the base and hopefully the diff will re-correct

Reply:
Looks like a lot of people are adding counters to combat these kind of problems, but I  like @tay1orjones suggests another option is to generate a unique ID. With this solution the table is getting a new ID on each re-render, which doesn't seem optimal.

Perhaps we isolate the fix to the TableHead.jsx file and reduce some table complexity while keeping the ID fixed while the component is alive? We could use lodash and the react state do something like `const [ checkboxID ] = useState(() => uniqueId('select-all-'))` in TableHead.jsx.

I did notice that the checkbox itself has the following comment

```
/* TODO: Replace checkbox with TableSelectAll component when onChange bug is fixed
                    https://github.com/IBM/carbon-components-react/issues/1088 */
```

The URL is dead now, but we should probably look into that before adding more workarounds with the current checkbox.

Reply:
Even further, could probably avoid state and just useMemo?
```js
const tableID = useMemo(() => `table-${Date.now()}-${tableId}`, [tableId]); // could use counter instead of Date.now
```

IIRC That comment is pretty old, from the initial implementation of the Table. I've opened https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1520 to address separately

Reply:
Yea, useMemo could work but it is not guaranteed that the ID will not change, according to the docs:
"In You may rely on useMemo as a performance optimization, not as a semantic guarantee"

Reply:
Seems to me that IF the ID is required for the table to operate correctly, then it should be required.  Hacking the id to avoid using a require prop seems wrong.